generators/arduino/procedures.js

```diff
@@ -44,8 +44,8 @@ function getCodeGenerators(Arduino) {
     }
 
     // Get return type
-    var returnType = "double";
-    //returnType = Arduino.getArduinoType_(returnType);
+    var returnType = block.type;
```

Comment:
block.type bevat de 'id' van het block, dus in dit geval `procedures_defreturn` of zoiets.

Op regel 28 wordt gekeken of er een returnValue is of niet. In leaphy programma's hebben we eigenlijk alleen maar return type 'void' (geen return type), of 'double'. Als je dus de if (returnValue) wat uitbreidt, kun je daar de returnType ("void" of "double") bepalen.